### ISSUE TITLE
Isn't the training loss, binary cross entropy loss?

### ISSUE THREAD

**Issue by meliscicek** [georgeretsi/pytorch-phocnet#2]
Since
loss = CosineLoss(size_average=False, use_sigmoid=True)
loss_val = loss(output, embedding)*args.batch_size
loss_val.backward()

Does this mean training is based on cosine loss?
As I understand from the paper, training loss is based on crossentropy.

**Comment by georgeretsi** [comment]
The training loss can be either binary cross entropy (BCE) or cosine. 
Both approaches are explored by Sebastian Sudholt at "Evaluating Word String Embeddings and Loss Functions for CNN-based Word Spotting".
In the code (train,py) BCE is selected as the predefined:

    loss_selection = 'BCE' # or 'cosine'
    if loss_selection == 'BCE':
        loss = nn.BCEWithLogitsLoss(size_average=True)
    elif loss_selection == 'cosine':
        loss = CosineLoss(size_average=False, use_sigmoid=True)

**Comment by meliscicek** [comment]
Thank you!
